plugin/src/generators/sentry/utils/generate-sentry-next.ts

```diff
@@ -0,0 +1,126 @@
+  createPrinter().printFile(
+    tsquery.map(
+      tsquery.ast(content),
+      'Identifier[name="nextConfig"] ~ ObjectLiteralExpression',
+      (node: ObjectLiteralExpression) => {
+        return createObjectLiteralExpression(
+          [
+            {
+              key: 'widenClientFileUpload',
+              initializer: factory.createTrue(),
+              comment:
+                'Upload a larger set of source maps for prettier stack traces (increases build time)',
+            },
+            {
+              key: 'transpileClientSDK',
+              initializer: factory.createTrue(),
+              comment:
+                'Transpiles SDK to be compatible with IE11 (increases bundle size)',
+            },
+            {
+              key: 'tunnelRoute',
+              initializer: factory.createStringLiteral('/monitoring'),
+              comment:
+                'Routes browser requests to Sentry through a Next.js rewrite to circumvent ad-blockers (increases server load)',
+            },
+            {
+              key: 'hideSourceMaps',
+              initializer: factory.createTrue(),
+              comment: 'Hides source maps from generated client bundles',
+            },
+            {
+              key: 'disableLogger',
+              initializer: factory.createTrue(),
+              comment:
+                'Automatically tree-shake Sentry logger statements to reduce bundle size',
+            },
+          ],
+          node.properties,
+        );
+      },
+      {
+        visitAllChildren: true,
+      },
+    ),
+  );
+
+const wrapIntoSentryConfig = (content: string): string => {
+  const withSentryWebpackPluginOptions = tsquery.replace(
+    content,
+    'ExpressionStatement:has(CallExpression Identifier[name="withNx"])',
+    (node) => {
+      return `
+      /**
+      * @type {import('@sentry/nextjs').SentryWebpackPluginOptions}
+      **/
+
+      const sentryWebpackPluginOptions = {
+        silent: true,
+        org: '',
+        project: 'web-next-js-client',
+        authToken: process.env.SENTRY_AUTH_TOKEN,
+      };
+
+      ${node.getText()}`;
+    },
+    {
+      visitAllChildren: true,
+    },
+  );
+
+  return tsquery.replace(
+    withSentryWebpackPluginOptions,
+    'CallExpression > CallExpression:has(Identifier[name="withNx"]) Identifier[name="nextConfig"]',
+    (node) => `withSentryConfig(${node.getText()}, sentryWebpackPluginOptions)`,
+    {
+      visitAllChildren: true,
+    },
+  );
+};
+
+export function generateSentryNext(
+  tree: Tree,
+  options: SentryGeneratorSchema,
+  projectRoot: string,
+) {
+  addDependenciesToPackageJson(tree, nextAppDependencies, {});
+
+  const nextConfigContent = tree
+    .read(`${projectRoot}/next.config.js`)
+    .toString();
+
+  const updatedNextConfigContent = wrapIntoSentryConfig(
+    modifyNextConfig(addRequiredImports(nextConfigContent)),
+  );
+
+  tree.write(`${projectRoot}/next.config.js`, updatedNextConfigContent);
```

Comment:
We could implement some utility function to shorten the code that performs such file updates. Very rough idea:

```ts
function updateFile(tree, filePath, updater) {
  // 1. Call tree.read to get file contents
  // 2. Call updater over the file contents
  // 3. Call tree.write to write the updated file contents
}
```